src/main/java/by/exadel/internship/service/impl/UserServiceImpl.java

```diff
@@ -149,4 +153,14 @@ public void updateTimeSlot(UUID userId, List<UserTimeSlotDTO> userTimeSlotDTOLis
         return mapper.map(users);
     }
 
+    @Override
+    public Set<UserDTO> getUsersBySkills(List<Skill> skills) {
+        List<String> skillsToString = skills.stream().map(Enum::toString).collect(Collectors.toList());
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to get set of user by skills : {}", skills);
+        List<User> usersWithSkills = userRepository.getUsersBySkills(skillsToString);
+        Set<User> newList = new HashSet<>(usersWithSkills);
```

Comment:
Why Set? Can you return List?

Reply:
Set because 1 User can have 2 request skills and in the List he we be added 2 times

Reply:
Try to use `distinct` in query.

Reply:
And I got 2 users the same ones

Reply:
Done, thanks